### DISCUSSION TITLE
Left join creating values of null

### DISCUSSION THREAD

**Discussion by milestones95** [supabase/supabase#2334] [Questions]
Hi, I am trying to do a left join with 2 tables:

writer_stories and left join the user_story by 'story_id'. I want to get all of the mismatches that have the story_name as null. Here is the query:
```sql
create or replace function "WriterStories"(writer uuid, user_id uuid)
returns table(story_id int, story_name varchar, us_story_name varchar) as $$
  select distinct
    ws.story_id, ws.story_name, us.story_name
  from writer_stories ws
  left join user_story us on ws.story_id = us.story_id  
  where ws.writer_id = writer;
$$ language sql;
```

For some reason, rows that have a story name in the database are now showing up in the query with the story_name as null. It doesn't exist in the db though. Any idea what's wrong with my query?

**Comment by dragarcia** [comment]
Hi @milestones95 👋 

Interesting query! I just want to clarify a few things to best answer your question:
> I want to get all of the mismatches that have the story_name as null

Which `story_name` are you referring to? Is it from the `writer_stories` table or from the `user_story` one?

> rows that have a story name in the database are now showing up in the query with the story_name as null

Likewise for this one.

Also, would you mind sharing more about the two tables? In particular their purposes, schemas, and a few sample rows from each one?

Thanks! 😄

**Reply by milestones95** [reply]
Which story_name are you referring to? Is it from the writer_stories table or from the user_story one?
- the user_story one

rows that have a story name in the database are now showing up in the query with the story_name as null
- user_story

The writer_stories table is a table of template stories where different names can be inserted into a story. Users can receive stories from writers with their name info inserted into the template. This story specific to the user goes into user_story table. 

The **writer_stories** table looks like this:
category_id |   | integer | int4 |  
story_id |   | integer | int4 |  
story_name |   | character varying | varchar |  
writer_id |   | uuid | uuid

**user_story table:**
category_id |   | integer | int4 |  
id |   | integer | int4 |  
order_id |   | uuid | uuid |  
read_time |   | integer | int4 |  
story_id |   | integer | int4 |  
story_name |   | character varying | varchar |  
user_id |   | uuid | uuid |  
writer_id |   | uuid | uuid

I want to see what stories in the writer table have not been sent to a specific user (user_id) before to make sure writers don't send duplicate stories to users.

**Reply by dragarcia** [reply]
Ah okay, got it!

> rows that have a story name in the database are now showing up in the query with the story_name as null

This should be expected then. The query would be returning all rows from the `writer_stories` table, regardless of whether there is a match in the `user_story` table. Hence the `null` values.

> I want to see what stories in the writer table have not been sent to a specific user (user_id) before to make sure writers don't send duplicate stories to users.

Given this, may I suggest an alternative underlying query to achieve this:
```sql
select
  *
from writer_stories
where story_id not in (
  select story_id
  from user_story us
  where us.user_id = user_id
);
```
